src/app/features/courses/courses.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input } from '@angular/core';
+import { CourseDataType } from '@app/shared/types/course.model';
+
+@Component({
+  selector: 'app-courses',
+  templateUrl: './courses.component.html',
+  styleUrls: ['./courses.component.scss']
+})
+export class CoursesComponent {
+  @Input() courses:CourseDataType[]=[];
+  showCourseInfo:CourseDataType | undefined=undefined;
```

Comment:
I suggest that `showCourseInfo` would mark a boolean type information, weather the course info in displayed or not. Combining `CourseDataType` with `undefined` seems a little overkill.

Reply:
i dont get this point. 
Here I need to know the whole data of the course which I want to show. -> boolean is not enough. I could make a separate boolean property for deciding if I need to show the courses list or only the information about one course, but I found that bit redundant. I would still need data. This way when I click on back I set it to undefined and when I get a show event with course id I show that course.   Is it wrong practice and I should use a boolean variable too?

Reply:
First thing first, I don't recall that it was a requirement to show the course info on list item click, but it will be addressed later with routing (where you will need to show the course info component and actual course content based on some route with course id).
What I find a little misleading is the variable name compared to the value it can hold. 'Show' suggests that it should be true or false (my pervious comment), however it holds CourseData or undefined. On the .html template it is confusing when used as an input for CourseInfoComponent: `<app-course-info [course]="showCourseInfo">`
I find it hard to understand and this can cause some issues later on in the app development. Best to address them as they arise.
For now, I think it will be fine to simply rename the variable! `courseInfo` will do just fine, which suggest that it represents an actual CourseData entity or undefined (if no course with given id in the course list)

Reply:
yes, you are right. It is better to use here courseInfo or courseInfoContent(data.. ) names to show it is the actual data. -> changed